src/datatype.ts

```diff
@@ -34,39 +34,27 @@ export class Datatype {
   number(
     options?: number | { min?: number; max?: number; precision?: number }
   ): number {
-    if (typeof options === 'number') {
-      options = { max: options };
-    }
-
-    options = options ?? {};
+    const opts = typeof options === 'number' ? { max: options } : options ?? {};
 
-    let max = 99999;
-    let min = 0;
-    let precision = 1;
-    if (typeof options.min === 'number') {
-      min = options.min;
-    }
-
-    if (typeof options.max === 'number') {
-      max = options.max;
-    }
+    const precision = typeof opts.precision === 'number' ? opts.precision : 1;
+    const min = typeof opts.min === 'number' ? opts.min : 0;
+    let max = typeof opts.max === 'number' ? opts.max : min + 99999;
```

Comment:
Could you swap lines a bit to reduce diff-changes?

```suggestion
    const min = typeof opts.min === 'number' ? opts.min : 0;
    let max = typeof opts.max === 'number' ? opts.max : min + 99999;
    const precision = typeof opts.precision === 'number' ? opts.precision : 1;
```

min and max need to be swapped because now max uses min